src/precice/impl/WatchIntegral.cpp

```diff
@@ -105,22 +105,22 @@ void WatchIntegral::exportIntegralData(
   }
 }
 
-Eigen::VectorXd WatchIntegral::calculateIntegral(const mesh::PtrData &data) const
+Eigen::VectorXd WatchIntegral::calculateIntegral(double time, const mesh::PtrData &data) const
 {
-  int                    dim    = data->getDimensions();
-  const Eigen::VectorXd &values = data->timeStepsStorage().last().sample.values;
-  Eigen::VectorXd        sum    = Eigen::VectorXd::Zero(dim);
+  int             dim    = data->getDimensions();
+  const auto      sample = data->timeStepsStorage().sample(time);
+  Eigen::VectorXd sum    = Eigen::VectorXd::Zero(dim);
 
   if (_mesh->edges().empty() || (not _isScalingOn)) {
     for (const auto &vertex : _mesh->vertices()) {
       int offset = vertex.getID() * dim;
       for (int i = 0; i < dim; i++) {
-        sum[i] += values[offset + i];
+        sum[i] += sample.values()[offset + i];
```

Comment:
Based on recent additions. Also below

```suggestion
        sum[i] += sample(offset + i);
```